client/me/purchases/remove-purchase/index.jsx

```diff
@@ -176,28 +355,273 @@ const RemovePurchase = React.createClass( {
+
+		const tooHardInput = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="tooHardInput"
+				id="tooHardInput"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'Where did you run into problems?' ) } />
+		);
+		reasons.tooHard = (
+			<FormLabel key="tooHard">
+				<FormRadio
+					name="tooHard"
+					value="tooHard"
+					checked={ 'tooHard' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'It was too hard to set up my site.' ) }</span>
+				{ 'tooHard' === this.state.questionOneRadio ? tooHardInput : null }
+			</FormLabel>
+		);
+
+		const didNotIncludeInput = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="didNotIncludeInput"
+				id="didNotIncludeInput"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'What are we missing that you need?' ) } />
+		);
+		reasons.didNotInclude = (
+			<FormLabel key="didNotInclude">
+				<FormRadio
+					name="didNotInclude"
+					value="didNotInclude"
+					checked={ 'didNotInclude' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'This upgrade didn\'t include what I needed.' ) }</span>
+				{ 'didNotInclude' === this.state.questionOneRadio ? didNotIncludeInput : null }
+			</FormLabel>
+		);
+
+		const onlyNeedFreeInput = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="onlyNeedFreeInput"
+				id="onlyNeedFreeInput"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'How can we improve our upgrades?' ) } />
+		);
+		reasons.onlyNeedFree = (
+			<FormLabel key="onlyNeedFree">
+				<FormRadio
+					name="onlyNeedFree"
+					value="onlyNeedFree"
+					checked={ 'onlyNeedFree' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'All I need is the free plan.' ) }</span>
+				{ 'onlyNeedFree' === this.state.questionOneRadio ? onlyNeedFreeInput : null }
+			</FormLabel>
+		);
+
+		const anotherReasonOneInput = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="anotherReasonOneInput"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				id="anotherReasonOneInput" />
+		);
+		reasons.anotherReasonOne = (
+			<FormLabel key="anotherReasonOne">
+				<FormRadio
+					name="anotherReasonOne"
+					value="anotherReasonOne"
+					checked={ 'anotherReasonOne' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'Another reason…' ) }</span>
+				{ 'anotherReasonOne' === this.state.questionOneRadio ? anotherReasonOneInput : null }
+			</FormLabel>
+		);
+
+		for ( let i in this.state.questionOneOrder ) {
+			orderedReasons.push( reasons[ this.state.questionOneOrder[ i ] ] );
```

Comment:
hmm... I think I lost a comment somehow, because I already mentioned this but I'm not seeing it in github anymore. This part can be cleaner if we use a functional approach. Less mutation, more hipstery. Like this:
```js
const { questionOneOrder } = this.state;
const orderedReasons = questionOneOrder.map( question => reasons[ question ] );
```